### ISSUE TITLE
[Alerting] dashboardUid and panelId not recognized when provided via API

### ISSUE THREAD

**Issue by odam06** [grafana/grafana#62352]
**What happened**:
When we provide an alert via the terraform provider, the alert provided does not recognize the ```__dashboardUid__``` and ```__panelId__``` annotations which is blocking screenshots. Annotations are present, but not added in there proper columns (replaced by NULL)

It's working when the alert is created from the UI but not from the API

Here is the DB entry of the same alert based on the two possibility:

From UI:
```
| 866 | 1 | annotations_test_ui | C | [{"refId":"A","queryType":"","relativeTimeRange":{"from":600,"to":0},"datasourceUid":"WwWHMI5Mz","model":{"dimensions":{},"editorMode":"code","expr":"prometheus_engine_queries","expression":"","hide":false,"id":"","intervalMs":1000,"label":"","legendFormat":"__auto","matchExact":true,"maxDataPoints":43200,"metricEditorMode":0,"metricName":"","metricQueryType":0,"namespace":"","period":"","queryMode":"Metrics","range":true,"refId":"A","region":"default","sqlExpression":"","statistic":"Average"}},{"refId":"B","queryType":"","relativeTimeRange":{"from":600,"to":0},"datasourceUid":"-100","model":{"conditions":[{"evaluator":{"params":[],"type":"gt"},"operator":{"type":"and"},"query":{"params":["B"]},"reducer":{"params":[],"type":"last"},"type":"query"}],"datasource":{"type":"__expr__","uid":"-100"},"expression":"A","hide":false,"intervalMs":1000,"maxDataPoints":43200,"reducer":"last","refId":"B","type":"reduce"}},{"refId":"C","queryType":"","relativeTimeRange":{"from":600,"to":0},"datasourceUid":"-100","model":{"conditions":[{"evaluator":{"params":[0],"type":"gt"},"operator":{"type":"and"},"query":{"params":["C"]},"reducer":{"params":[],"type":"last"},"type":"query"}],"datasource":{"type":"__expr__","uid":"-100"},"expression":"B","hide":false,"intervalMs":1000,"maxDataPoints":43200,"refId":"C","type":"threshold"}}] | 2023-01-27 14:42:50 | 60 | 1 | 9vmn-9T4k | 9vcs771Wk | annotations_test_ui | NoData | Error | 300000000000 | {"__dashboardUid__":"ZXg90kJGz","__panelId__":"22"} | {} | ZXg90kJGz | 22 | 1 |
```

From TF:
```
| 867 | 1 | annotations_test_tf | C | [{"refId":"A","queryType":"metrics","relativeTimeRange":{"from":600,"to":0},"datasourceUid":"WwWHMI5Mz","model":{"editorMode":"code","examplar":false,"expr":"prometheus_engine_queries","hide":false,"instant":false,"interval":null,"intervalMs":1000,"legendFormat":"__auto","maxDataPoints":43200,"range":true,"refId":"A"}},{"refId":"B","queryType":"","relativeTimeRange":{"from":0,"to":0},"datasourceUid":"-100","model":{"conditions":[],"downsampler":null,"expression":"A","intervalMs":1000,"maxDataPoints":43200,"reducer":"last","refId":"B","settings":{"mode":"","replaceWithValue":null},"type":"reduce","upsampler":null,"window":null}},{"refId":"C","queryType":"","relativeTimeRange":{"from":0,"to":0},"datasourceUid":"-100","model":{"conditions":[],"downsampler":null,"expression":"$B \u003e 0","intervalMs":1000,"maxDataPoints":43200,"reducer":null,"refId":"C","settings":null,"type":"math","upsampler":null,"window":null}}] | 2023-01-27 15:53:26 | 900 | 1 | PAdVs9oVk  | 9vcs771Wk | annotations_test_tf | OK | Error | 0 | {"__dashboardUid__":"ZXg90kJGz","__panelId__":"22"} | null | NULL | NULL | 0 |
```

**What you expected to happen**:
When an alert is provided via API, the ```__dashboardUid__``` and ```__panelId__``` are recognized and recorded correctly in the DB

**How to reproduce it (as minimally and precisely as possible)**:
We have reproduced it by using a simple alert like this one:

``` hcl
resource "grafana_rule_group" "annotations_test_tf" {
  folder_uid       = "9vcs771Wk"
  interval_seconds = 900
  name             = "annotations_test_tf"
  org_id           = 1

  rule {
    annotations    = { "__dashboardUid__" : "ZXg90kJGz", "__panelId__" : "22" }
    condition      = "C"
    exec_err_state = "Error"
    for            = "0s"
    name           = "annotations_test_tf"
    no_data_state  = "OK"

    data {
      datasource_uid = "WwWHMI5Mz"
      model = jsonencode(
        {
          editorMode    = "code"
          examplar      = false
          expr          = "prometheus_engine_queries"
          hide          = false
          instant       = false
          interval      = null
          intervalMs    = 1000
          legendFormat  = "__auto"
          maxDataPoints = 43200
          range         = true
          refId         = "A"
        }
      )
      query_type = "metrics"
      ref_id     = "A"

      relative_time_range {
        from = 600
        to   = 0
      }
    }
    data {
      datasource_uid = "-100"
      model = jsonencode(
        {
          conditions    = []
          downsampler   = null
          expression    = "A"
          intervalMs    = 1000
          maxDataPoints = 43200
          reducer       = "last"
          refId         = "B"
          settings = {
            mode             = ""
            replaceWithValue = null
          }
          type      = "reduce"
          upsampler = null
          window    = null
        }
      )
      ref_id = "B"

      relative_time_range {
        from = 0
        to   = 0
      }
    }
    data {
      datasource_uid = "-100"
      model = jsonencode(
        {
          conditions    = []
          downsampler   = null
          expression    = "$B > 0"
          intervalMs    = 1000
          maxDataPoints = 43200
          reducer       = null
          refId         = "C"
          settings      = null
          type          = "math"
          upsampler     = null
          window        = null
        }
      )
      ref_id = "C"

      relative_time_range {
        from = 0
        to   = 0
      }
    }
  }
}
```

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.3.1
- Data source type & version: ALL
- Terraform grafana provider: 1.34

**Comment by jantman** [comment]
This is impacting me as well. It essentially renders the alert provisioning API useless for me, as alerts provisioned via API (terraform) both lack screenshots in the alert message, and the "heartbeat" indicator on the panel indicating that an alert has been configured.

**Comment by JacobsonMT** [comment]
This was [fixed in v9.4.0](https://github.com/grafana/grafana/pull/60374) and has been backported to [v9.1.x](https://github.com/grafana/grafana/pull/60442), [9.2.x](https://github.com/grafana/grafana/pull/60443), and [v9.3.x](https://github.com/grafana/grafana/pull/60444)
